specific/cmo_purchase/models/purchase.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+from openerp.exceptions import ValidationError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    project_id = fields.Many2one(
+        'project.project',
+        string='Project Name',
+        ondelete='restrict',
+    )
+    order_ref = fields.Many2one(
+        'sale.order',
+        string='Quotation Number',
+        ondelete='restrict',
+    )
+    event_date_description = fields.Char(
+        string='Event Date',
+        size=250,
+    )
+    venue_description = fields.Char(
+        string='Venue',
+        size=250,
+    )
+    po_type_id = fields.Many2one(
+        'purchase.order.type.config',
+        string='PO Type',
+        required=True,
+    )
+    po_project = fields.Boolean(
+        string='PO Project',
+        related='po_type_id.po_project',
+    )
+    approve_ids = fields.Many2one(
+        'hr.employee',
+        string='PO Approve',
+        required=True,
+    )
+    operating_unit_id = fields.Many2one(
+        change_default=True,
+    )
+
+    @api.onchange('po_type_id')
+    def _onchange_po_type_id(self):
+        self.project_id = False
+        self.order_ref = False
+        self.event_date_description = False
+        self.venue_description = False
+        self.order_line = False
+        self.invoice_method = self.po_type_id and \
```

Comment:
In API v8, if we sure the filed is many2one, we con't need to check for False.
Here is enough,
`... = self.po_type_id.invoice_method or False`